proof/refine/AARCH64/ArchInvsDefs_H.thy

```diff
@@ -39,123 +36,20 @@ end
-  "itcbIPCBuffer (tcb_to_itcb' tcb) = tcbIPCBuffer tcb"
-  "itcbBoundNotification (tcb_to_itcb' tcb) = tcbBoundNotification tcb"
-  "itcbPriority (tcb_to_itcb' tcb) = tcbPriority tcb"
-  "itcbFault (tcb_to_itcb' tcb) = tcbFault tcb"
-  "itcbTimeSlice (tcb_to_itcb' tcb) = tcbTimeSlice tcb"
-  "itcbMCP (tcb_to_itcb' tcb) = tcbMCP tcb"
-  by (auto simp: tcb_to_itcb'_def)
-
-definition pred_tcb_at' :: "(itcb' \<Rightarrow> 'a) \<Rightarrow> ('a \<Rightarrow> bool) \<Rightarrow> machine_word \<Rightarrow> kernel_state \<Rightarrow> bool"
-  where
-  "pred_tcb_at' proj test \<equiv> obj_at' (\<lambda>ko. test (proj (tcb_to_itcb' ko)))"
-
-abbreviation st_tcb_at' :: "(thread_state \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "st_tcb_at' \<equiv> pred_tcb_at' itcbState"
-
-abbreviation bound_tcb_at' :: "(obj_ref option \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "bound_tcb_at' \<equiv> pred_tcb_at' itcbBoundNotification"
-
-abbreviation mcpriority_tcb_at' :: "(priority \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "mcpriority_tcb_at' \<equiv> pred_tcb_at' itcbMCP"
-
-lemma st_tcb_at'_def:
-  "st_tcb_at' test \<equiv> obj_at' (test \<circ> tcbState)"
-  by (simp add: pred_tcb_at'_def o_def)
-
-text \<open> cte with property at \<close>
-definition cte_wp_at' :: "(cte \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "cte_wp_at' P p s \<equiv> \<exists>cte::cte. fst (getObject p s) = {(cte,s)} \<and> P cte"
-
-abbreviation cte_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "cte_at' \<equiv> cte_wp_at' \<top>"
-
-abbreviation tcb_of' :: "kernel_object \<Rightarrow> tcb option" where
-  "tcb_of' \<equiv> projectKO_opt"
-
-abbreviation tcbs_of' :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> tcb option" where
-  "tcbs_of' s \<equiv> ksPSpace s |> tcb_of'"
-
-abbreviation tcbSchedPrevs_of :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> obj_ref option" where
-   "tcbSchedPrevs_of s \<equiv> tcbs_of' s |> tcbSchedPrev"
-
-abbreviation tcbSchedNexts_of :: "kernel_state \<Rightarrow> obj_ref \<Rightarrow> obj_ref option" where
-  "tcbSchedNexts_of s \<equiv> tcbs_of' s |> tcbSchedNext"
-
-abbreviation sym_heap_sched_pointers :: "global.kernel_state \<Rightarrow> bool" where
-  "sym_heap_sched_pointers s \<equiv> sym_heap (tcbSchedNexts_of s) (tcbSchedPrevs_of s)"
+(* FIXME arch-split: currently global in Invariants_H
+definition max_ipc_words :: machine_word where
+  "max_ipc_words \<equiv> capTransferDataSize + msgMaxLength + msgMaxExtraCaps + 2" *)
 
 definition tcb_cte_cases :: "machine_word \<rightharpoonup> ((tcb \<Rightarrow> cte) \<times> ((cte \<Rightarrow> cte) \<Rightarrow> tcb \<Rightarrow> tcb))" where
  "tcb_cte_cases \<equiv> [ 0 << cteSizeBits \<mapsto> (tcbCTable, tcbCTable_update),
```

Comment:
I would have expected this one to be a generic definition. Do we we have architectures that add TCB cap slots?

Reply:
No, but other than AARCH64 and RISCV64, they do this:
```
184: "tcb_cte_cases \<equiv> [ 0 \<mapsto> (tcbCTable, tcbCTable_update),
185-                   16 \<mapsto> (tcbVTable, tcbVTable_update),
186-                   32 \<mapsto> (tcbReply, tcbReply_update),
187-                   48 \<mapsto> (tcbCaller, tcbCaller_update),
188-                   64 \<mapsto> (tcbIPCBufferFrame, tcbIPCBufferFrame_update) ]"
```